src/Mesh/Gmsh_FileIO.cpp

```diff
@@ -1340,11 +1340,11 @@ void Gmsh_FileIO::update_quadratic_hex_IEN( const int &index_3d )
     eIEN[domain_index][nlocbas * ee + 19] = temp15;
     if (nlocbas == 27)
     {
-      const double temp20 = eIEN[domain_index][27 * ee + 20];
-      const double temp21 = eIEN[domain_index][27 * ee + 21];
-      const double temp22 = eIEN[domain_index][27 * ee + 22];
-      const double temp23 = eIEN[domain_index][27 * ee + 23];
-      const double temp24 = eIEN[domain_index][27 * ee + 24];
+      const int temp20 = eIEN[domain_index][27 * ee + 20];
+      const int temp21 = eIEN[domain_index][27 * ee + 21];
+      const int temp22 = eIEN[domain_index][27 * ee + 22];
+      const int temp23 = eIEN[domain_index][27 * ee + 23];
+      const int temp24 = eIEN[domain_index][27 * ee + 24];
```

Comment:
std::swap can be applicable here and avoid such type of errors easily.
https://en.cppreference.com/w/cpp/algorithm/swap

@Goldeninscriptions @Syj-0525

Reply:
maybe not :(

the reordering here is more complicated.